### ISSUE TITLE
Bug: Auto-scroll doesn't work on Chrome

### ISSUE THREAD

**Issue by tanyxp** [One-com/knockout-bottomsup#1]
We have used your knockout binding for auto scrolling, and it works great on Safari, Edge, and Firefox. For some reason, on Chrome (Windows) it auto-scroll won't work.
Any thoughts on why this is happening?

**Comment by papandreou** [comment]
I can't reproduce that. Using BrowserStack I tried opening [the demo page](http://one-com.github.io/knockout-bottomsup/examples/) in Chrome 56, 57, and 58 on Windows 8.1 and 10. Seems like the auto-scrolling works fine, the `foo <n>` entries keep appearing at the bottom of the containers.

Can you share some more details about the setting where it doesn't work, ideally a stripped down, runnable example?

**Comment by tanyxp** [comment]
Below is a gif of the demo page. The gif was taken on a Windows 10 64-bit computer with Chrome 58. On this same computer, the demo page works with Edge and with Firefox. I'm not sure what the issue is.
How can we provide you with more information to help resolve this issue?

![Demo Page](https://media.giphy.com/media/3oKIPnvG6HZAUN2m8U/giphy.gif)

**Comment by papandreou** [comment]
I assume the bug means that the comparison at [this line](https://github.com/One-com/knockout-bottomsup/blob/461fa2ee098b707c2f408a82f2e21e3f09f57b32/lib/knockout.bottomsUp.js#L22) comes out as false. Could you try adding some logging right before that so we can see what the values of `element.scrollHeight`, `element.scrollTop`, and `element.clientHeight` the first 5 times the `beforeChangeSubscription` is fired? That might help narrow down where the problem comes from.

**Comment by tanyxp** [comment]
First of all, thanks for the help so far. I appreciate it.
I added logging to the demo page with the three values you requested. This logging was placed in the `beforeChangeSubscription`.

```
foo 0
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 0
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 1
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 1
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 2
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 2
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 3
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 3
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 4
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 4
scrollHeight: 150
scrollTop: 0
clientHeight: 150

foo 5
scrollHeight: 184
scrollTop: 33.599998474121094
clientHeight: 150

foo 5
scrollHeight: 184
scrollTop: 33.599998474121094
clientHeight: 150

foo 6
scrollHeight: 221
scrollTop: 33.599998474121094
clientHeight: 150

foo 6
scrollHeight: 221
scrollTop: 33.599998474121094
clientHeight: 150

foo 7
scrollHeight: 258
scrollTop: 33.599998474121094
clientHeight: 150

foo 7
scrollHeight: 258
scrollTop: 33.599998474121094
clientHeight: 150

foo 8
scrollHeight: 295
scrollTop: 33.599998474121094
clientHeight: 150

foo 8
scrollHeight: 295
scrollTop: 33.599998474121094
clientHeight: 150

foo 9
scrollHeight: 332
scrollTop: 33.599998474121094
clientHeight: 150

foo 9
scrollHeight: 332
scrollTop: 33.599998474121094
clientHeight: 150

foo 10
scrollHeight: 369
scrollTop: 33.599998474121094
clientHeight: 150

foo 10
scrollHeight: 369
scrollTop: 33.599998474121094
clientHeight: 150

foo 11
scrollHeight: 406
scrollTop: 33.599998474121094
clientHeight: 150

foo 11
scrollHeight: 406
scrollTop: 33.599998474121094
clientHeight: 150

foo 12
scrollHeight: 443
scrollTop: 33.599998474121094
clientHeight: 150

foo 12
scrollHeight: 443
scrollTop: 33.599998474121094
clientHeight: 150
```

**Comment by papandreou** [comment]
A `scrollTop` of 33.599998474121094, huh? :). Could there be some weird zoom settings in place?

In this case it seems like it would be fixed by:

```diff
-                   isAtBottom = element.scrollHeight - element.scrollTop <= element.clientHeight;
+                   isAtBottom = Math.floor(element.scrollHeight - element.scrollTop) <= element.clientHeight;
```

... but maybe we should build in a tolerance of a pixel or two.

**Comment by papandreou** [comment]
Put in a tolerance of 2 pixels and released 1.0.1. Please reopen if there's still a problem.

Thanks for reporting this issue!

**Comment by tanyxp** [comment]
Thank you so much for the help! It worked beautifully! I really appreciate it!
